doc: nrf: releases: Clarify supplicant switch to kernel heap (#20108)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-2.8.0.rst

  * The WPA supplicant is now switched to Zephyr upstream's fork instead of |NCS|.
  * The WPA supplicant now uses ``kernel heap`` instead of ``application (libc) heap``.

>     - This means there is an increase in the kernel heap usage but a corresponding decrease in the application heap usage.

Reviewer: ```suggestion
    This means there is an increase in the kernel heap usage but a corresponding decrease in the application heap usage.
```

>     - Please note that this means there is an increase in kernel heap usage but a corresponding decrease in application heap usage.

Reviewer: ```suggestion
    This means there is an increase in the kernel heap usage but a corresponding decrease in the application heap usage.
```
